### PR TITLE
✨ Add a flag for branch name

### PR DESCRIPTION
**Pull request by afranke**
Signed-off-by: Alexandre Franke <alexandre.franke@gmail.com>

### CODE REVIEW

File: gha-webhook-listener.py [hunk 1]
```diff
@@ -291,6 +300,12 @@ def tidy_extract_directory(target_dir, cleanup_dir, versions_to_keep):
         ), required=True,
     )
 
+    parser.add_argument(
+        "--branch-name", dest="branch_name", default="master", help=(
+            "Name of the branch to deploy from."
```
Comment (richvdh) [inline gha-webhook-listener.py:305]: not so much to deploy from:, rather to only accept pushes from?

File: gha-webhook-listener.py [hunk 2]
```diff
@@ -137,6 +138,14 @@ def on_receive_poke():
         abort(400, "Unrecognised event")
         return
 
+    workflow_status = incoming_json["workflow_run"]["head_branch"]
+    if workflow_status == arg_branch_name:
+        logger.info(f"Workflow was for branch {arg_branch_name}")
+    else:
+        logger.info("Rejecting '%s' event", event)
+        abort(400, "Wrong branch")
```
Comment (richvdh) [inline gha-webhook-listener.py:146]: we should just ignore it, and return a 200 rather than a 400?

File: gha-webhook-listener.py [hunk 3]
```diff
@@ -291,6 +300,12 @@ def tidy_extract_directory(target_dir, cleanup_dir, versions_to_keep):
         ), required=True,
     )
 
+    parser.add_argument(
+        "--branch-name", dest="branch_name", default="master", help=(
+            "Name of the branch to deploy from."
+        ), required=True,
```
Comment (richvdh) [inline gha-webhook-listener.py:306]: what does it mean to set `required` when there is a default?

**Review by richvdh — CHANGES_REQUESTED** [review]

File: gha-webhook-listener.py (hunk 2 above)
Comment (afranke) [reply gha-webhook-listener.py:146]: Would this still use `abort`?

File: gha-webhook-listener.py (hunk 3 above)
Comment (afranke) [reply gha-webhook-listener.py:306]: It means I’m a script kiddie and copied from the wrong lines.

File: gha-webhook-listener.py (hunk 2 above)
Comment (richvdh) [reply gha-webhook-listener.py:146]: I don't think so. You just return the body you want to return in the response, same as line 212:

```
    return jsonify({})
```

File: gha-webhook-listener.py [hunk 4]
```diff
@@ -291,6 +300,13 @@ def tidy_extract_directory(target_dir, cleanup_dir, versions_to_keep):
         ), required=True,
     )
 
+    parser.add_argument(
+        "--branch-name", dest="branch_name", default="master", help=(
+            "Name of the branch to only accept pushes from."
```
Comment (richvdh) [inline gha-webhook-listener.py:305]:
```suggestion
            "Branch to accept build notifications for. Notifications for other branches will be ignored. "
```

**Review by richvdh — APPROVED** [review]

**Pull request merged**